### PR TITLE
feat: main container + router

### PR DESCRIPTION
**Pull request by 1MrSmith**

### CODE REVIEW

File: js_machine_front/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "js_machine_front",
-  "version": "0.1.0",
+  "version": "0.2.1",
   "private": true,
   "dependencies": {
     "@types/jest": "24.0.15",
```
Comment (vharadkou) [inline js_machine_front/package.json]: please relocate @types dependencies to devDependencies, react-scripts and typescript too

File: js_machine_front/src/components/MainContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import { Route, Switch, Redirect } from 'react-router';
+import { NavLink } from 'react-router-dom';
+import '../styles/route.css';
+
+// tslint:disable-next-line: typedef
```
Comment (vharadkou) [inline js_machine_front/src/components/MainContainer.tsx]: please don't ignore tslint warnings

File: js_machine_front/src/components/MainContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    render(): JSX.Element {
+        return (
+          <div>
+            <nav className="nav">
+              <span className="nav__link"><NavLink exact={true} activeClassName="activeLink" to="/about">О НАС</NavLink></span>
+              <span className="nav__link"><NavLink exact={true} activeClassName="activeLink" to="/news">НОВОСТИ</NavLink></span>
+              <span className="nav__link"><NavLink exact={true} activeClassName="activeLink" to="/events">СОБЫТИЯ</NavLink></span>
+              <span className="nav__link"><NavLink exact={true} activeClassName="activeLink" to="/partners">ПАРТНЕРЫ</NavLink></span>
+            </nav>
+            {/* <MainComponent {...this.props as P}/>
```
Comment (vharadkou) [inline js_machine_front/src/components/MainContainer.tsx]: unused code?

File: js_machine_front/src/index.tsx [hunk 4]
```diff
@@ -1,12 +1,22 @@
...
-serviceWorker.unregister();
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import * as serviceWorker from './serviceWorker';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { MainContainer } from './components/MainContainer';
+import { About } from './components/About';
+
+const App = MainContainer(About, About, About, About, About);
```
Comment (vharadkou) [inline js_machine_front/src/index.tsx]: need to provide a better solution

**Review by vharadkou — CHANGES_REQUESTED** [review]

File: js_machine_front/package.json (hunk 1 above)
Comment (1MrSmith) [reply js_machine_front/package.json]: fix

File: js_machine_front/src/components/MainContainer.tsx (hunk 3 above)
Comment (1MrSmith) [reply js_machine_front/src/components/MainContainer.tsx]: It has been done for scroll components but I am not sure it's landing page or spa

File: js_machine_front/src/styles/route.css [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+.nav {
+    position: absolute;
+    right: 3vh;
+    top: 8vh;
+}
+
+.nav__link {
+    padding-right: 8vh;
+}
+a {
```
Comment (mkinitcpio) [inline js_machine_front/src/styles/route.css:14]: Remove style for "a" tags. Better use class.

File: js_machine_front/src/components/MainContainer.tsx (hunk 2 above)
Comment (1MrSmith) [reply js_machine_front/src/components/MainContainer.tsx]: ok

File: js_machine_front/src/index.tsx (hunk 4 above)
Comment (1MrSmith) [reply js_machine_front/src/index.tsx]: fix

File: js_machine_front/src/styles/route.css (hunk 5 above)
Comment (1MrSmith) [reply js_machine_front/src/styles/route.css:14]: yes

**Comment by vharadkou** [comment]
[Landing Page (Main container)](https://trello.com/c/2qi1rFpP/12-landing-page-main-container)

File: js_machine_front/src/components/About.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React, { memo } from 'react';
```
Comment (vharadkou) [inline js_machine_front/src/components/About.tsx]: please relocate all components related to routing into 'routes' folder

**Review by vharadkou — CHANGES_REQUESTED** [review]

File: js_machine_front/src/App.tsx [hunk 7]
```diff
@@ -1,26 +1,15 @@
...
-export default App;
+import React from 'react';
+import './App.css';
+import { Routes } from 'components/Routes';
+import { NavBar } from 'components/NavBar';
+
+const App: React.FC = () => {
+  return (
+    <div className="App">
+      <Routes />
```
Comment (vharadkou) [inline js_machine_front/src/App.tsx]: please do not create separate components for routes, they all must be in App.tsx

**Review by vharadkou — CHANGES_REQUESTED** [review]

File: js_machine_front/src/components/About.tsx (hunk 6 above)
Comment (1MrSmith) [reply js_machine_front/src/components/About.tsx]: fix

File: js_machine_front/src/App.tsx (hunk 7 above)
Comment (1MrSmith) [reply js_machine_front/src/App.tsx]: fix

**Review by vharadkou — APPROVED** [review]

**Pull request merged**